BurnOutSharp/PackerType/InnoSetup.cs

```diff
@@ -49,39 +57,41 @@ public string CheckContents(string file, byte[] fileContent, bool includePositio
         {
-            byte[] signature = new ArraySegment<byte>(fileContent, 0x30, 12).ToArray();
-
-            // "rDlPtS02" + (char)0x87 + "eVx"
-            if (signature.SequenceEqual( new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x32, 0x87, 0x65, 0x56, 0x78 }))
-                return "1.2.10";
+            var matchers = new List<ContentMatchSet>
+            {
+                // "rDlPtS02" + (char)0x87 + "eVx"
+                new ContentMatchSet(new byte?[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x32, 0x87, 0x65, 0x56, 0x78 }, "1.2.16 or earlier"),
+            };
 
-            // "rDlPtS04" + (char)0x87 + "eVx"
-            else if (signature.SequenceEqual(new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x34, 0x87, 0x65, 0x56, 0x78 }))
-                return "4.0.0";
+            string match = MatchUtil.GetFirstMatch(file, fileContent, matchers, false);
+            return match ?? "Unknown 1.X";
+        }
 
-            // "rDlPtS05" + (char)0x87 + "eVx"
-            else if (signature.SequenceEqual(new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x35, 0x87, 0x65, 0x56, 0x78 }))
-                return "4.0.3";
+        public static string GetVersion(string file, byte[] fileContent, List<int> positions)
+        {
+            try
+            {
+                int index = positions[0];
+                index += 23;
 
-            // "rDlPtS06" + (char)0x87 + "eVx"
-            else if (signature.SequenceEqual(new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x36, 0x87, 0x65, 0x56, 0x78 }))
-                return "4.0.10";
+                var versionBytes = new ReadOnlySpan<byte>(fileContent, index, 16).ToArray();
+                var onlyVersion = versionBytes.TakeWhile(b => b != ')').ToArray();
 
-            // "rDlPtS07" + (char)0x87 + "eVx"
-            else if (signature.SequenceEqual(new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0x30, 0x37, 0x87, 0x65, 0x56, 0x78 }))
-                return "4.1.6";
+                index += onlyVersion.Length + 2;
+                var unicodeBytes = new ReadOnlySpan<byte>(fileContent, index, 3).ToArray();
 
-            // "rDlPtS" + (char)0xcd + (char)0xe6 + (char)0xd7 + "{" + (char)0x0b + "*"
-            else if (signature.SequenceEqual(new byte[] { 0x72, 0x44, 0x6C, 0x50, 0x74, 0x53, 0xCD, 0xE6, 0xD7, 0x7B, 0x0b, 0x2A }))
-                return "5.1.5";
 
-            // "nS5W7dT" + (char)0x83 + (char)0xaa + (char)0x1b + (char)0x0f + "j"
-            else if (signature.SequenceEqual(new byte[] { 0x6E, 0x53, 0x35, 0x57, 0x37, 0x64, 0x54, 0x83, 0xAA, 0x1B, 0x0F, 0x6A }))
-                return "5.1.5";
+                if (unicodeBytes.SequenceEqual(new byte[] { 0x28, 0x75, 0x29 }))
+                    return (Encoding.ASCII.GetString(onlyVersion) + " (Unicode)");
 
-            return string.Empty;
+                return Encoding.ASCII.GetString(onlyVersion);
```

Comment:
Since you're using this in two places, you can do something like `string version = Encoding.ASCII.GetString(onlyVersion);` above the if